presto-spi/src/main/java/com/facebook/presto/spi/PageSinkProperty.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi;
+
+public class PageSinkProperty
+{
+    public static final PageSinkProperty DEFAULT_PAGE_SINK_PROPERTY = new PageSinkProperty.Builder().build();
+
+    private final boolean lifespanCommitRequired;
+
+    private PageSinkProperty(boolean lifespanCommitRequired)
+    {
+        this.lifespanCommitRequired = lifespanCommitRequired;
+    }
+
+    public boolean isLifespanCommitRequired()
```

Comment:
Oh no... `Lifespan` is a Presto engine concept I guess....

Reply:
Yeah I'm struggling to find a name that means "partial" commit. "partial" by itself is kind of vague :( Maybe we should discuss offline? cc @wenleix

Reply:
@highker : That's a good point. Connector shouldn't know about `lifespan`. 

Now I think about it, the key on the connector side is `data partition` (not Hive partition, but Hive bucket). From engine perspective, to execute lifespan X, it reads the partition X from input tables, and write the computation result to partition X in output table.

In fact, for the input side, this is called `partition handle`: https://github.com/prestodb/presto/blob/81118abcad44bb7275da975b55553ac2485cb785/presto-spi/src/main/java/com/facebook/presto/spi/ConnectorSplitSource.java#L27

And on the write side, it's really to commit a "partition".  (Although this name has different meaning in Hive). What about `isPartitionGranularityCommitRequired`?

Reply:
Why not just simple `isPartitionCommitRequired`? We already have the concepts clash in Hive connector. e.g.: `getNextBatch(ConnectorPartitionHandle partitionHandle`, where `partitionHandle` actually refers to a bucket.